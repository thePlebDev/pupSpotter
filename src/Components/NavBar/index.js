import React,{useState} from 'react';
import styled from 'styled-components';
import PetsIcon from '@material-ui/icons/Pets';
import HomeIcon from '@material-ui/icons/Home';
import HelpOutlineIcon from '@material-ui/icons/HelpOutline';
import {Link} from 'react-router-dom';
import {Tab} from '../../Assets/SideBarStyles';
import LoginStatus from '../LoginStatus';
import PropTypes from 'prop-types';

const HomeTab = styled(Tab)`
  left:0px;
  border-radius:4px;
  width:215px;
  display:flex;
  justify-content:space-between;
  align-items:center;
  opacity:0.8;
  transform:translateX(-150px);
  cursor:pointer;
  z-index:99999999999999999;

  &:hover{
    opacity:1;
    transform:translateX(-30px);:
  }
`
const SpotTab = styled(HomeTab)`
  top:15%;
`
const LoginTab = styled(HomeTab)`
  top:30%;
`
const AboutTab = styled(HomeTab)`
  top:45%;
`



const Text = styled.div`
  font-size:1.4em;
  color:white;
  margin-left:60px;
  cursor:pointer;
`


const NavBar = ({loggedIn,setLoggedIn})=> {
  const [show,setShow] = useState(true)
  return (
    <div style={{position:'absolute',left:0,height:'100vh'}}>
    <div />
      <HomeTab className="tab">
        <Link to="/">
          <Text>Home</Text>
        </Link>
          <HomeIcon style={{fontSize:'60',color:'white',cursor:'pointer'}} onClick={()=>setShow(!show)}  />

      </HomeTab>
      <SpotTab className="tab">
        <Link to="/pupSpotting">
          <Text>Spot</Text>
        </Link>
        <PetsIcon style={{fontSize:'60',color:'white',cursor:'pointer'}} onClick={()=>setShow(!show)}  />
      </SpotTab>

      <LoginTab className="tab">
          <LoginStatus loggedIn={loggedIn} setLoggedIn={setLoggedIn} />
      </LoginTab>

      <AboutTab className="tab">
        <Text>About</Text>
        <HelpOutlineIcon style={{fontSize:'60',color:'white',cursor:'pointer'}} onClick={()=>setShow(!show)}  />
      </AboutTab>

    </div>
  );
}

NavBar.propTypes ={
  loggedIn:PropTypes.bool,
  setLoggedIn:PropTypes.func
}

export default NavBar
