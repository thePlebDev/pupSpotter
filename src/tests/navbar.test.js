import React from 'react';
import NavBar from '../Components/NavBar';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import {BrowserRouter as Router} from 'react-router-dom';


Enzyme.configure({
  adapter: new EnzymeAdapter()
})

describe('testing component',()=>{
  it('should render with no errors',()=>{
    const component = shallow(<NavBar/>)
    const wrapper = component.find('.tab');
    expect(wrapper.length).toBe(4)
  })
  it('creating the snapshot',()=>{
    const NavBarComponent = renderer.create(<Router><NavBar /></Router>).toJSON();
    expect(NavBarComponent).toMatchSnapshot();
  })
})
