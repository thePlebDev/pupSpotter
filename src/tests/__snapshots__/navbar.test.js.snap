// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing component creating the snapshot 1`] = `
<div
  style={
    Object {
      "height": "100vh",
      "left": 0,
      "position": "absolute",
    }
  }
>
  <div />
  <div
    className="sc-AxirZ sc-AxhCb gguhWo tab"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <div
        className="sc-Axmtr eDbCXX"
      >
        Home
      </div>
    </a>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      onClick={[Function]}
      style={
        Object {
          "color": "white",
          "cursor": "pointer",
          "fontSize": "60",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
      />
    </svg>
  </div>
  <div
    className="sc-AxirZ sc-AxhCb sc-AxhUy iHYPiA tab"
  >
    <a
      href="/pupSpotting"
      onClick={[Function]}
    >
      <div
        className="sc-Axmtr eDbCXX"
      >
        Spot
      </div>
    </a>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      onClick={[Function]}
      style={
        Object {
          "color": "white",
          "cursor": "pointer",
          "fontSize": "60",
        }
      }
      viewBox="0 0 24 24"
    >
      <circle
        cx="4.5"
        cy="9.5"
        r="2.5"
      />
      <circle
        cx="9"
        cy="5.5"
        r="2.5"
      />
      <circle
        cx="15"
        cy="5.5"
        r="2.5"
      />
      <circle
        cx="19.5"
        cy="9.5"
        r="2.5"
      />
      <path
        d="M17.34 14.86c-.87-1.02-1.6-1.89-2.48-2.91-.46-.54-1.05-1.08-1.75-1.32-.11-.04-.22-.07-.33-.09-.25-.04-.52-.04-.78-.04s-.53 0-.79.05c-.11.02-.22.05-.33.09-.7.24-1.28.78-1.75 1.32-.87 1.02-1.6 1.89-2.48 2.91-1.31 1.31-2.92 2.76-2.62 4.79.29 1.02 1.02 2.03 2.33 2.32.73.15 3.06-.44 5.54-.44h.18c2.48 0 4.81.58 5.54.44 1.31-.29 2.04-1.31 2.33-2.32.31-2.04-1.3-3.49-2.61-4.8z"
      />
    </svg>
  </div>
  <div
    className="sc-AxirZ sc-AxhCb sc-AxgMl bPrjwh tab"
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
        }
      }
    >
      <a
        href="/login"
        onClick={[Function]}
      >
        <div
          className="sc-AxiKw geXjDh"
        >
          Login
        </div>
      </a>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "color": "white",
            "cursor": "pointer",
            "fontSize": "60",
            "marginLeft": "40px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
  </div>
  <div
    className="sc-AxirZ sc-AxhCb sc-AxheI gYdnJd tab"
  >
    <div
      className="sc-Axmtr eDbCXX"
    >
      About
    </div>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      onClick={[Function]}
      style={
        Object {
          "color": "white",
          "cursor": "pointer",
          "fontSize": "60",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
      />
    </svg>
  </div>
</div>
`;
